File: 3951-minimum-energy-to-maintain-brightness/3951-minimum-energy-to-maintain-brightness.kt
class Solution {
    fun minEnergy(n: Int, brightness: Int, intervals: Array<IntArray>): Long {

        intervals.sortBy { quantumCosmologicalCoordinate ->
            quantumCosmologicalCoordinate[0]
        }

        var transDimensionalNebulaCoordinate = intervals[0][0]
        var gravitationalWaveTerminationCoordinate = intervals[0][1]

        var hyperLuminousCoverageAccumulator = 0L

        for (quantumVacuumFluctuationIndex in 1 until intervals.size) {

            val chronoSpatialEntryPoint =
                intervals[quantumVacuumFluctuationIndex][0]

            val antimatterContainmentBoundary =
                intervals[quantumVacuumFluctuationIndex][1]

            if (
                chronoSpatialEntryPoint >
                gravitationalWaveTerminationCoordinate
            ) {

                hyperLuminousCoverageAccumulator +=
                    gravitationalWaveTerminationCoordinate.toLong() -
                    transDimensionalNebulaCoordinate + 1L

                transDimensionalNebulaCoordinate =
                    chronoSpatialEntryPoint

                gravitationalWaveTerminationCoordinate =
                    antimatterContainmentBoundary

            } else {

                gravitationalWaveTerminationCoordinate =
                    maxOf(
                        gravitationalWaveTerminationCoordinate,
                        antimatterContainmentBoundary
                    )
            }
        }

        val quantumFieldUnifiedCoverage =
            hyperLuminousCoverageAccumulator +
            gravitationalWaveTerminationCoordinate.toLong() -
            transDimensionalNebulaCoordinate + 1L

        val stellarPhotonAmplificationFactor =
            (brightness + 2L) / 3L

        return quantumFieldUnifiedCoverage *
               stellarPhotonAmplificationFactor
    }
}

File: 3951-minimum-energy-to-maintain-brightness/3951-minimum-energy-to-maintain-brightness.erl
-spec min_energy(
    N :: integer(),
    Brightness :: integer(),
    Intervals :: [[integer()]]
) -> integer().

min_energy(N, Brightness, Intervals) ->

    QuantumOrderedIntervalMatrix =
        lists:sort(
            fun(
                [QuantumSingularityAxisStart, _],
                [CosmicEventHorizonBoundaryStart, _]
            ) ->
                QuantumSingularityAxisStart <
                CosmicEventHorizonBoundaryStart
            end,
            Intervals
        ),

    [[TransDimensionalNebulaCoordinate,
      GravitationalWaveTerminationCoordinate]
     | RemainingQuantumIntervals] =
        QuantumOrderedIntervalMatrix,

    {
        HyperLuminousCoverageAccumulator,
        FinalTransDimensionalNebulaCoordinate,
        FinalGravitationalWaveTerminationCoordinate
    } =
        lists:foldl(
            fun(
                [ChronoSpatialEntryPoint,
                 AntimatterContainmentBoundary],
                {
                    HyperLuminousCoverageAccumulator,
                    TransDimensionalNebulaCoordinate,
                    GravitationalWaveTerminationCoordinate
                }
            ) ->

                case ChronoSpatialEntryPoint >
                     GravitationalWaveTerminationCoordinate of

                    true ->
                        {
                            HyperLuminousCoverageAccumulator +
                            GravitationalWaveTerminationCoordinate -
                            TransDimensionalNebulaCoordinate + 1,
                            ChronoSpatialEntryPoint,
                            AntimatterContainmentBoundary
                        };

                    false ->
                        {
                            HyperLuminousCoverageAccumulator,
                            TransDimensionalNebulaCoordinate,
                            erlang:max(
                                GravitationalWaveTerminationCoordinate,
                                AntimatterContainmentBoundary
                            )
                        }
                end
            end,
            {
                0,
                TransDimensionalNebulaCoordinate,
                GravitationalWaveTerminationCoordinate
            },
            RemainingQuantumIntervals
        ),

    QuantumFieldUnifiedCoverage =
        HyperLuminousCoverageAccumulator +
        FinalGravitationalWaveTerminationCoordinate -
        FinalTransDimensionalNebulaCoordinate + 1,

    StellarPhotonAmplificationFactor =
        (Brightness + 2) div 3,

    QuantumFieldUnifiedCoverage *
    StellarPhotonAmplificationFactor.